### DISCUSSION TITLE
Segment visitors by logged in state

### DISCUSSION THREAD

**Discussion by metmarkosaric** [plausible/analytics#800] [Anything Else]
" “Signup” conversion doesn’t mean much if you can’t segment our existing members.

100 visitors, 10x "SignUp" goals = 10% conv., right. But, if 50x visitors are already "members" then only 50x visitors might be expected to sign up... conv. is actually 20%. Without being able to segment out folks who will/should never hit the goal, they're not as useful.

a lot of our traffic is "signed-in" traffic... so they can never convert / hit our most important goal (to "sign up")... (same problem again to measure freemium ➜ premium .. just the numbers are smaller!!"

([request via twitter](https://twitter.com/jimbomorrison/status/1368609541982715913))

**Comment by Vigasaurus** [comment]
This seems to just be #320, no? With that, people should be able to attach a logged in state (via the cookie etc) in pageview events, and then filter by that

**Reply by ukutaht** [reply]
More or less. I think the logged-in state would better fit on the session level rather than pageview level. It describes the whole session not just one pageview.

So I think we can tease 2 independent features our from these. One for custom props on pageview level and one for session level.

**Comment by luxaritas** [comment]
Was just about to raise an issue asking about what the current best practice is for monitoring something like this. It sounds like either we need to have a way to associate data with a session (if this was a separate function call, that would also allow for avoiding using manual pageviews when wanting to associate data that is not unique per pageview) or a way to filter properties by "occured sometime this session" or "did not occur sometime this session".

This may be valuable enough (its looking like knowing our signup conversion rate is going to be important!) for me to contribute a PR depending on complexity (and maybe with a little guidance - unfortunately I've never touched elixer before :see_no_evil:)

**Reply by metmarkosaric** [reply]
hi @luxaritas! the best option would be to use our custom properties for pageviews. see https://plausible.io/docs/custom-pageview-props

**Reply by luxaritas** [reply]
Hey @metmarkosaric - that is indeed what I'm doing now, but afaik there isn't a way to filter by "this occurred sometime in the user's session" vs "this never occurred in the user's session", only filtering individual pageviews where it did occur

**Reply by metmarkosaric** [reply]
ah i see what you mean @luxaritas! props are on the pageview level only and not on the session level. it will still give you some useful data but we're about to release funnels next week which should be more relevant for conversion rate. stay tuned #171

**Reply by luxaritas** [reply]
I'm still not sure if funnels would solve this use case - doesn't that only track if some event positively happens over the user's lifetime? It would need to both 1) let you filter down other stats to just users who _didnt_ complete a goal and 2) allow for filtering on page view properties that happened at some time during the user's lifetime (events are insufficient as if we just fire an event on login, tomorrow they'll be shown as a new user who hasn't logged in because the login event happened on a different analytics user due to the salt change)
